### DISCUSSION TITLE
DW 3 Validation appears not to be enabled...

### DISCUSSION THREAD

**Discussion by jim-trux** [dropwizard/dropwizard#7110] [Q&A]
We have been using DW 2+ for awhile now and we have created some services with DW 3 and I am running into an issue where the REST controller is not validating inputs. 

For example a create method with the `javax @Valid annotation`

```java 

    @POST
    @Path("/")
    @Operation(description="Create default adjustment from the given prototype")
    @RolesAllowed({"SYSTEM", "ADMIN", "CUSTOMER_ADMIN"})
    public Adjustment create( @NotNull @Valid AdjustmentDTO prototype, @Auth User actingUser) {

        Adjustment result = null;

        try {
            prototype.setCreatedBy(actingUser.getUserId());
            result = service.create(prototype);
        }
        catch (DomainException e)
        {
            throwProblem(getBaseUri(request), getLanguage(request), Status.INTERNAL_SERVER_ERROR, e);
        }
        catch (RuntimeException e)
        {
            throwProblem(Status.INTERNAL_SERVER_ERROR, e);
        }

        return result;
    }
```

The AdjustmentDTO has javax constraints 

```java
@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder(builderMethodName = "of", toBuilder = true)
public class AdjustmentDTO
{
    @NotNull
    private WorkWorld workItemWorld;

    @Length(max = 100)
    private String  productExternalId;

    @NotNull
    private Long originId;

    @NotNull
    private Long adjustmentTypeId;

    @NotNull
    @Digits(integer = 12, fraction = 5)
    private BigDecimal totalQuantity;

}
```

The only response we get back when the constraints are violated is 400 rather than 422 as documented.

The interesting thing for me is that it looks like GW3 is documented with javax annotations but the dependencies such as DW metrics includes the jakarta annotations and jakarata validations api. 

What am I missing?

Thanks, 

Jim Mochel

**Comment by joschi** [comment]
There are various tests making sure that the validation of JAX-RS/Jersey resources is working:
- https://github.com/dropwizard/dropwizard/blob/v3.0.0/dropwizard-jersey/src/test/java/io/dropwizard/jersey/validation/ConstraintViolationExceptionMapperTest.java
- https://github.com/dropwizard/dropwizard/blob/v3.0.0/dropwizard-jersey/src/test/java/io/dropwizard/other/RestInterface.java

What's in the (DEBUG?) logs of your application when sending an invalid request to the respective resources?

> DW metrics includes the jakarta annotations and jakarata validations api.

Could you please elaborate? Dropwizard Metrics isn't using the Jakarta Bean Validation API at all nor did it use the old Java Bean Validation API.
